### PR TITLE
Deesharma/exttypescli

### PR DESCRIPTION
**Pull request by deeksha345**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az k8s-extension extension-types show
az k8s-extension extension-types list
az k8s-extension extension-types list-by-location
az k8s-extension extension-types list-versions

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8s-extension

**Comment by zhoxing-ms** [comment]
> ModuleNotFoundError: No module named 'azext_k8s_configuration.vendored_sdks.v2021_09_01.aio.operations._cluster_extension_type_operations'

This CI issue is caused by the absence of `_cluster_extension_type_operations` file in directory [src/k8s-extension/azext_k8s_extension/vendored_sdks/v2021_09_01/aio/operations](https://github.com/Azure/azure-cli-extensions/tree/654355f4ec9f070371df68b99c62bd897950b77a/src/k8s-extension/azext_k8s_extension/vendored_sdks/v2021_09_01/aio/operations)

This may be caused by you did not specify the api-version for `cf_k8s_extension_operation`

https://github.com/Azure/azure-cli-extensions/blob/654355f4ec9f070371df68b99c62bd897950b77a/src/k8s-extension/azext_k8s_extension/_client_factory.py#L15-L16

So there should still be a reference to api-version `v2021_09_01` in the Python SDK
https://github.com/Azure/azure-cli-extensions/blob/654355f4ec9f070371df68b99c62bd897950b77a/src/k8s-extension/azext_k8s_extension/vendored_sdks/aio/_source_control_configuration_client.py#L221-L222

File: src/k8s-configuration/azext_k8s_configuration/vendored_sdks/v2021_09_01/aio/operations/__init__.py [hunk 1]
```diff
@@ -8,10 +8,20 @@
 
 from ._extensions_operations import ExtensionsOperations
 from ._operation_status_operations import OperationStatusOperations
+from ._cluster_extension_type_operations import ClusterExtensionTypeOperations
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/vendored_sdks/v2021_09_01/aio/operations/__init__.py:11]: CI problem should be introduced by this line. May I ask why do you want to modify this file?

Comment (deeksha345) [reply src/k8s-configuration/azext_k8s_configuration/vendored_sdks/v2021_09_01/aio/operations/__init__.py:11]: I did not make that change I do not know why it is included in this.

Comment (zhoxing-ms) [reply src/k8s-configuration/azext_k8s_configuration/vendored_sdks/v2021_09_01/aio/operations/__init__.py:11]: OK, please revert this change in your PR

Comment (deeksha345) [reply src/k8s-configuration/azext_k8s_configuration/vendored_sdks/v2021_09_01/aio/operations/__init__.py]: I have reverted that change and now all checks are passing. Thank you for finding the issue!

File: src/k8s-configuration/azext_k8s_configuration/vendored_sdks/v2021_09_01/aio/operations/__init__.py [hunk 2]
```diff
@@ -13,5 +13,10 @@
 __all__ = [
     'ExtensionsOperations',
     'OperationStatusOperations',
+    'ClusterExtensionTypeOperations',
+    'ClusterExtensionTypesOperations',
+    'ExtensionTypeVersionsOperations',
+    'LocationExtensionTypesOperations',
+    'SourceControlConfigurationsOperations',
     'Operations',
 ]
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/vendored_sdks/v2021_09_01/aio/operations/__init__.py:22]: May I ask why do you need to modify the file in `k8s-configuration` folder?

File: src/k8s-extension/azext_k8s_extension/_client_factory.py [hunk 3]
```diff
@@ -7,15 +7,31 @@
...
+def cf_k8s_cluster_extension_type_operation(cli_ctx, _):
+    return cf_k8s_extension(cli_ctx, api_version='2022-01-15-preview').cluster_extension_type
+
+
+def cf_k8s_location_extension_types_operation(cli_ctx, _):
+    return cf_k8s_extension(cli_ctx, api_version='2022-01-15-preview').location_extension_types
+
+
+def cf_k8s_extension_type_versions_operation(cli_ctx, _):
+    return cf_k8s_extension(cli_ctx, api_version='2022-01-15-preview').extension_type_versions
```
Comment (zhoxing-ms) [inline src/k8s-extension/azext_k8s_extension/_client_factory.py:32]: If the `api_version` used by these operations need to be consistent, could we abstract `2022-01-15-preview` into a reusable constant

File: src/k8s-extension/azext_k8s_extension/_client_factory.py [hunk 4]
```diff
@@ -5,17 +5,33 @@
 
 from azure.cli.core.commands.client_factory import get_mgmt_service_client
 from azure.cli.core.profiles import ResourceType
+from . import consts
 
-
-def cf_k8s_extension(cli_ctx, *_):
+def cf_k8s_extension(cli_ctx, **kwargs):
```
Comment (zhoxing-ms) [inline src/k8s-extension/azext_k8s_extension/_client_factory.py:10]: > Running flake8 on extensions...
> /home/vsts/work/1/s/src/k8s-extension/azext_k8s_extension/_client_factory.py:10:1: E302 expected 2 blank lines, found 1
1     E302 expected 2 blank lines, found 1
Please fix the CI issue

```suggestion

def cf_k8s_extension(cli_ctx, **kwargs):
```

**Comment by zhoxing-ms** [comment]
Please write the description of these change into the `HISTORY.rst`. In addition, if you want to release a new extension for this PR, please also modify the `setup.py`

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**